airflow/models/dagrun.py

```diff
@@ -491,6 +491,25 @@ def is_backfill(self):
             self.run_id.startswith(f"{DagRunType.BACKFILL_JOB.value}")
         )
 
+    @classmethod
+    @provide_session
+    def get_latest_run(cls, dag_id, session):
+        """Returns the latest DagRun for a given dag_id"""
+        subquery = (
+            session
+            .query(func.max(cls.execution_date).label('execution_date'))
+            .filter(cls.dag_id == dag_id)
+            .subquery()
+        )
+        dagrun = (
```

Comment:
can these two queries be combined into one using order by and limit?

Reply:
they can, but since we don't have index on execution_date, order_by will translate to sort on postgres, and thus it may be more expensive (at least on postgres), 
```
airflow=# explain select max(execution_date) from dag_run;
                          QUERY PLAN                           
---------------------------------------------------------------
 Aggregate  (cost=10.75..10.76 rows=1 width=8)
   ->  Seq Scan on dag_run  (cost=0.00..10.60 rows=60 width=8)
(2 rows)

airflow=# explain select execution_date from dag_run order by execution_date desc limit 1;
                             QUERY PLAN                              
---------------------------------------------------------------------
 Limit  (cost=10.90..10.90 rows=1 width=8)
   ->  Sort  (cost=10.90..11.05 rows=60 width=8)
         Sort Key: execution_date DESC
         ->  Seq Scan on dag_run  (cost=0.00..10.60 rows=60 width=8)
(4 rows)
```

Reply:
ha, ok, i didn't know execution_date column is not indexed, that's surprising. in that case, subquery is better.